pkg/cmd/extensions/manager.go

```diff
@@ -99,17 +98,39 @@ func (m *Manager) list(includeMetadata bool) []extensions.Extension {
 			continue
 		}
 		var remoteURL string
+		var updatable bool
 		if gitExe != "" {
-			stdout := bytes.Buffer{}
-			cmd := m.newCommand(gitExe, "--git-dir="+filepath.Join(dir, f.Name(), ".git"), "config", "remote.origin.url")
-			cmd.Stdout = &stdout
-			if err := cmd.Run(); err == nil {
-				remoteURL = strings.TrimSpace(stdout.String())
-			}
+			func() {
+				gitDir := "--git-dir=" + filepath.Join(dir, f.Name(), ".git")
+
+				cmd := m.newCommand(gitExe, gitDir, "config", "remote.origin.url")
+				url, err := cmd.Output()
+				if err != nil {
+					return
+				}
+				remoteURL = strings.TrimSpace(string(url))
+
+				cmd = m.newCommand(gitExe, gitDir, "rev-parse", "origin")
```

Comment:
Does `rev-parse` consult the network? If not, how would a local git repository potentially know about new commits pushed to the remote repository?

Reply:
`rev-parse` will use the network to retrieve the remote SHA.

Reply:
Are you sure? I've just tried it in an out-of-date checkout:
```
$ git rev-parse origin
ebdcf0c26ed0039762dca01844ce8175c0d02755
$ git pull
....
$ git rev-parse origin
8a3cfa5d0d001bac6a06dc107e0fa2943876bed0
```

Reply:
Well I was sure when I wrote this... I did some sort of testing around this locally to make sure it worked... but I am unable to reproduce what I did and now I think you are correct that `rev-parse` won't actually use the network to get the remote SHA 🤦

Reply:
I am going to switch to using [ls-remote](https://git-scm.com/docs/git-ls-remote.html).